services/feedz/feedz.service.spec.js

```diff
@@ -0,0 +1,215 @@
+        })
+      )
+        .to.throw(NotFound)
+        .with.property('prettyMessage', 'package not found')
+    })
+  })
+
+  describe('tryHandleError', function () {
+    it('re-throws error on non-404 response', async function () {
+      try {
+        await serviceInstance.tryHandleError({
+          err: error({ statusCode: 400 }),
+        })
+        expect.fail('Expected to throw')
+      } catch (e) {
+        expect(e).to.be.an.instanceof(InvalidResponse)
+        expect(e.prettyMessage).to.equal('fake error')
+      }
+    })
+
+    it('re-throws error on other errors', async function () {
+      try {
+        await serviceInstance.tryHandleError({
+          err: new Error('other error'),
+        })
+        expect.fail('Expected to throw')
+      } catch (e) {
+        expect(e).to.be.an.instanceof(Error)
+        expect(e.message).to.equal('other error')
+      }
+    })
+
+    it('re-throws error on non-JSON body', async function () {
+      try {
+        await serviceInstance.tryHandleError({
+          err: error({ statusCode: 404 }),
+        })
+        expect.fail('Expected to throw')
+      } catch (e) {
+        expect(e).to.be.an.instanceof(InvalidResponse)
+        expect(e.prettyMessage).to.equal('fake error')
+      }
+    })
+    it('re-throws error when body does not match schema', async function () {
+      try {
+        await serviceInstance.tryHandleError({
+          err: error({ statusCode: 404, content: '{}' }),
+        })
+        expect.fail('Expected to throw')
+      } catch (e) {
+        expect(e).to.be.an.instanceof(InvalidResponse)
+        expect(e.prettyMessage).to.equal('fake error')
+      }
+    })
+
+    it('re-throws error when message does not match template', async function () {
+      try {
+        await serviceInstance.tryHandleError({
+          err: error({ statusCode: 404, message: 'unknown message' }),
+        })
+        expect.fail('Expected to throw')
+      } catch (e) {
+        expect(e).to.be.an.instanceof(InvalidResponse)
+        expect(e.prettyMessage).to.equal('fake error')
+      }
+    })
+
+    it('throws repository NotFound when message matches "org/repo not found" template', async function () {
+      try {
+        await serviceInstance.tryHandleError({
+          err: error({ statusCode: 404, message: 'Feed a for b not found' }),
+        })
+        expect.fail('Expected to throw')
+      } catch (e) {
+        expect(e).to.be.an.instanceof(NotFound)
+        expect(e.prettyMessage).to.equal('repository not found')
+      }
+    })
+
+    it('throws package NotFound when the content is empty', async function () {
+      try {
+        await serviceInstance.tryHandleError({
+          err: error({ statusCode: 404, content: '' }),
+        })
+        expect.fail('Expected to throw')
+      } catch (e) {
+        expect(e).to.be.an.instanceof(NotFound)
+        expect(e.prettyMessage).to.equal('package not found')
+      }
+    })
+  })
```

Comment:
Let's remove these tests.

The service class infra/parent classes are already tested extensively elsewhere. We don't want to re-test parent class members from unit tests in all the extending child classes as that just adds maintenance overhead on our end and makes the test suite ever so slower (especially we started doing this in _every_ child class).

Reply:
Sorry, ignore this for now, I completely missed the modifications to the service class that these are testing